key.go

```diff
@@ -376,14 +388,14 @@ func PubKeyFromMultibaseString(s string) (*PubKey, error) {
 		// secpEc.NewPublicKey accepts any valid encoding, while we
 		// explicitly want compressed, so use the explicit point
 		// decompression routine.
-		p, err := secp.NewIdentityPoint().SetCompressedBytes(raw)
+		p, err := secp.NewIdentityPoint().SetBytes(raw)
```

Comment:
If you're ok with accepting both compressed and uncompressed encodings, then you can just call secec.NewPublicKey, instead of deserializing the point.

Reply:
Ah, okay.
I think I am happy to be more lenient in what we accept here, seems safe enough?

Reply:
I have no strong opinions here, as long as the spec isn't explicit about behavior, though "only accept the format that is outputted" was why I was more strict in the first place.

If deserialization does not fail, the point is guaranteed to be a point on the curve (ie: a valid public key), so as long as the fact that there will be 2 different encodings that represent the same public key isn't a problem, it's fine.